_inc/lib/jetpack-wpes-query-builder/jetpack-wpes-query-builder.php

```diff
@@ -338,4 +338,33 @@ public function build_aggregation() {
 		return $aggregations;
 	}
 
+	/**
+	 * Assemble the 'suggest' portion of an ES query, using a phrase suggester. Note that this is experimental and
+	 * only works with English tokens.
+	 *
+	 * @param string $query The query text being searched.
+	 *
+	 * @return array An experimental suggest query.
+	 */
+	public function build_experimental_suggest( $query ) {
```

Comment:
I like the idea of this being in the query parser at some point (rather than here in the builder). I'm not sure we know enough about what a "generic" function should look like though, and once it is in this API I really don't want to change it in case anyone uses it.

Also non-obvious, this class is used internally for a bunch of search applications beyond JP Search. So I would say let's move this into the main jetpack search class as a private method for now.